### PR TITLE
fix: MenuPopover should stay open for submenu even when not openOnHover

### LINKED ISSUE #30394: [Bug]: openOnHover 'false' closes submenus when the mouse courses moves out of the submenu item trigger
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/sandbox/nice-orla-gdz2jt

### Bug Description

Open the [codepen with repro](https://codesandbox.io/p/sandbox/nice-orla-gdz2jt)

This is a forked [codepen from the NestedMenu section from storybook](https://react.fluentui.dev/?path=/docs/components-menu-menu--default&globals=storybook_fluentui-react-addon_dir:ltr#nested-submenus) where just the `openOnHover` property on the submenu is set to `false`.

1. Click Toggle Menu
2. Click on "Preferences" and leave the mouse cursor on the "Preferences"
3. Move the mouse on the submenu

## Actual Behavior
The submenu closes

## Expected Behavior
The submenu stays open as long as the user didn't move or click away from it.

The origin of the stack trace is here: https://github.com/microsoft/fluentui/blob/be3a3d59d5c30f75d55c739548d576594ddf5c53/packages/react-components/react-menu/src/utils/useOnMenuEnter.ts#L33

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by ling1726**
Currently submenus will only stay open when the mouse enters the MenuPopover when `openOnHover` is set. The assumption was that all submenus would generally be opened with hover.

Updates the check on mouse enter for the MenuPopover so that if the MenuPopover is part of submenu, it should also stay open on mouse enter.

Fixes #30394

### CODE REVIEW

File: packages/react-components/react-menu/src/components/MenuPopover/useMenuPopover.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export const useMenuPopover_unstable = (props: MenuPopoverProps, ref: React.Ref<
   );
   const { onMouseEnter: onMouseEnterOriginal, onKeyDown: onKeyDownOriginal } = rootProps;
   rootProps.onMouseEnter = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
-    if (openOnHover) {
+    if (openOnHover || isSubmenu) {
```
Comment (bsunderhus) [inline packages/react-components/react-menu/src/components/MenuPopover/useMenuPopover.ts:77]: nit: add some tests to ensure this behaviour?!

**Review by bsunderhus — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Avatar-mount_base.html">632</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Avatar-mount.html">624</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Button-mount_base.html">301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Button-mount.html">307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Field-mount_base.html">1120</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Field-mount.html">1143</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProvider-mount_base.html">694</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProvider-mount.html">685</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProviderWithTheme-mount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProviderWithTheme-mount.html">89</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/MakeStyles-mount_base.html">870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/MakeStyles-mount.html">849</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Persona-mount_base.html">1775</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/Persona-mount.html">1705</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/SpinButton-mount_base.html">1389</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30486/perf-test-react-components/SpinButton-mount.html">1479</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                               Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> |  `217.02 kB`<br />`61.894 kB` | `217.025 kB`<br />`61.899 kB` | `5 B` <br />`5 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `149.308 kB`<br />`45.469 kB` | `149.313 kB`<br />`45.473 kB` | `5 B` <br />`4 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `151.994 kB`<br />`45.981 kB` | `151.999 kB`<br />`45.984 kB` | `5 B` <br />`3 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                |         Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `70.688 kB`<br />`20.464 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>               | `43.173 kB`<br />`14.298 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                            |   `7.902 kB`<br />`2.586 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1b8ade8a67f75ec4007f54a52a94f61879f6b215'>1b8ade8a67f75ec4007f54a52a94f61879f6b215</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**